pay_ios/ios/Classes/PaymentHandler.swift

```diff
@@ -179,14 +194,20 @@ extension PaymentHandler: PKPaymentAuthorizationControllerDelegate {
     // Return the result back to the channel
     self.paymentResult(String(decoding: paymentResultData, as: UTF8.self))
     
-    paymentStatus = .success
-    completion(PKPaymentAuthorizationResult(status: paymentStatus, errors: nil))
+    paymentHandlerStatus = .authorized
+
```

Comment:
I think removing this line would be useful to conceptually connect calling the `completion` handler and the status update.